Example/Example/Views/Pages/AuthPage.xaml.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Xamarin.Forms;
+
+using Xamarin.Forms.Xaml;
+namespace Example.Views.Pages
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class AuthPage : ContentPage
+    {
+        public AuthPage()
+        {
+            InitializeComponent();
+        }
+
+        public void OnAuthenticateClicked(object sender, EventArgs args)
```

Comment:
Can you cast the argument here directly? `(Button sender, EventArgs args)`

Reply:
Nope. A runtime error is thrown if I change the signature.